src/components/verify/document-renderer.tsx

```diff
@@ -1,16 +1,17 @@
 import { FrameActions, FrameConnector, HostActionsHandler } from "@govtechsg/decentralized-renderer-react-components";
-import { getData, WrappedDocument } from "@govtechsg/open-attestation";
+import { getData, v2, WrappedDocument } from "@govtechsg/open-attestation";
 import React, { useState, useCallback, useEffect, useMemo } from "react";
 import { Tabs } from "../tabs";
 
 interface DocumentRendererProps {
-  rawDocument: WrappedDocument;
+  rawDocument: WrappedDocument<v2.OpenAttestationDocument>;
 }
 
 export const DocumentRenderer: React.FunctionComponent<DocumentRendererProps> = ({
   rawDocument,
 }: DocumentRendererProps) => {
-  const document = useMemo(() => getData(rawDocument), [rawDocument]);
+  const document: v2.OpenAttestationDocument = useMemo(() => getData(rawDocument), [rawDocument]);
+  const template: any = document.$template;
```

Comment:
not sure how to type this, as `document.$template.url` was complaining that document might be undefined

Reply:
- dont type document => it should be automatically inferred.
- from https://schema.openattestation.com/2.0/schema.json, template might be a string, an object or might not exist so typescript is correct
  - let's remove this line `const template: any = document.$template;`
  - let's add a check before rendering this component that there is a URL. Can check how it's handled in OC => https://github.com/OpenCerts/opencerts-website/blob/master/src/components/DecentralisedTemplateRenderer/DecentralisedRenderer.tsx#L119
  - let's ignore the string case :)

Reply:
^ this helps 👍